spezi-module/onboarding/src/main/java/edu/stanford/spezikt/spezi_module/onboarding/onboarding/OnboardingViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package edu.stanford.spezikt.spezi_module.onboarding.onboarding
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import edu.stanford.spezikt.coroutines.di.Dispatching
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class OnboardingViewModel @Inject internal constructor(
+    private val repository: OnboardingRepository,
+    @Dispatching.IO private val scope: CoroutineScope,
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(OnboardingUiState())
+    val uiState = _uiState.asStateFlow()
+
+    init {
+        init()
+    }
+
+    fun onAction(action: Action) {
+        _uiState.update {
+            when (action) {
+                is Action.UpdateArea -> {
+                    val newAreas = action.areas
+                    it.copy(areas = newAreas)
+                }
+
+                Action.OnLearnMoreClicked -> TODO()
+                Action.ClearError -> it.copy(error = null)
+            }
+        }
+    }
+
+    private fun init() {
+        scope.launch {
+            val result = repository.getAreas()
```

Comment:
Related to the onboarding repository comment (one single method suggestion). here you would invoke that method, and based on the result either display the content, or introduce an error state in the screen (consider a sealed interface for `OnboardingUiState` - `Error` and `Success(OnboardingData)`